labs/authentication-passport/README.md

```diff
@@ -0,0 +1,185 @@
+# Passport Strategy Adapter
+
+_Important: We suggest users understand LoopBack's authentication system[Link
+TBD](some loopback.io link) before using this module_
+
+This is an adapter module created for plugging in
+[`passport`](https://www.npmjs.com/package/passport) base strategies to the
+authentication system in `@loopback/authentication@2.x`.
+
+## Installation
+
+```sh
+npm i @loopback/passport-adapter --save
+```
+
+## Background
+
+`@loopback/authentication@2.x` allows users to register authentication
+strategies that implement the interface
+[`AuthenticationStrategy`](https://apidocs.strongloop.com/@loopback%2fdocs/authentication.html#AuthenticationStrategy)
+
+Since `AuthenticationStrategy` describes a strategy with different contracts
+than the passport
+[`Strategy`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/passport/index.d.ts#L79),
+and we'd like to support the existing 500+ community passport strategies, an
+**adapter class** is created in this package to convert a passport strategy to
+the one that LoopBack 4 authentication system wants.
+
+## Usage
+
+### Simply Usage
+
+1. Created an instance of the passport strategy
+
+Take the basic strategy exported from
+[`passport-http`](https://github.com/jaredhanson/passport-http) as an example,
+first create an instance of the basic strategy with your `verify` function.
+
+```ts
+function verify(username: string, password: string, cb: Function) {
+  users.find(username, password, cb);
+}
+const basicStrategy = new BasicStrategy(verify);
+```
+
+It's the similar configuration as you do when adding a strategy to a `passport`
+by calling `passport.use()`.
+
+But don't provide any passport specific options since we don't support the
```

Comment:
> But don't provide any passport specific options since we don't support the session manager.

Can you give an example of what you mean here? And what happens if that passport strategy requires a session manager? Will it not work?

Reply:
I doubled checked the usage of `passport-http`, and realize the passport specific options are only provided when the strategy is invoked by `passport.authenticate()`, like https://github.com/jaredhanson/passport-http#authenticate-requests.
So, since we don't involve `passport` in this PR, we don't introduce that concern either. I will remove ~"But don't provide any passport specific options since we don't support the session manager."~